### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kimitaWanjohi**
in this pr. 
- I added the person, teacher, and student classes.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Person
+  attr_accessor :id, :name, :age
```
Comment (brytebee) [inline person.rb:2]: The `Id` attribute should only be retrieved by a getter, a user should not be able to alter the id of your data.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def can_use_services
+    if of_age || @parent_permission
+      true
+    else
+      false
+    end
+  end
+
+  public :can_use_services
```
Comment (brytebee) [inline person.rb:25]: - [ ] **Optional:** Conventionally, all public methods are defined, and then the `private` keyword is used and every method defined below it is set to private by Ruby.

```Ruby
class ExampleClass
  # attributes
  
  #initialize method

  # public methods
  private

  # private methods
end
```

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @kimitaWanjohi 👋,

##

### Highlights💡
- Great understanding of the Ruby style guide.✔️
- Awesome class definitions.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by brytebee** [comment]
@kimitaWanjohi For future projects kindly use a feature name that doesn't seem like the develop branch. For example, this branch could be called `classes`, `ruby_classes`, `feature_classes` or any name of choice. Your name could be confusing in a bigger project with many collaborators.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+class Person
+  attr_accessor :name, :age
+
```
Comment (Gambit142) [inline person.rb:3]: - [ ] Great work ensuring that the `id` attribute can't be written or altered by outside calls as requested by the previous reviewer. However, the `id` attribute can't be read of gotten. Kindly implement this change and make the id attribute accessible. You can use a getter method for this `attr_reader :id`. 👍 😎 You can learn more about the setters and getters method by reading this [article](https://www.geeksforgeeks.org/ruby-getters-and-setters-method/). 💻 🖱️

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @kimitaWanjohi 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- All linters checks passed ✔️ 
- Created `Person`, `Student` and `Teacher` classes ✔️ 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @kimitaWanjohi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Defined Person, Student & Teacher classes ✔️
- Well Structured code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**